e2e/test/scenarios/question/offset.cy.spec.ts

```diff
@@ -60,240 +67,415 @@ describe("scenarios > question > offset", () => {
+
+      enterCustomColumnDetails({ formula: suffix });
+      cy.realPress("Tab");
+
+      popover().within(() => {
+        cy.findByText("OFFSET in a custom expression requires a sort order");
+        cy.button("Done").should("be.disabled");
+        cy.button("Cancel").click();
+      });
+
+      cy.button("Sort").click();
+      popover().findByText("ID").click();
+      getNotebookStep("expression").icon("add").click();
+      enterCustomColumnDetails({ formula: expression });
+      cy.realPress("Tab");
 
-    createQuestion({ query }, { visitQuestion: true });
+      popover().within(() => {
+        cy.button("Done").should("be.disabled");
 
-    verifyQuestionError(
-      "Window function requires either breakouts or order by in the query",
-    );
-  });
+        cy.findByPlaceholderText("Something nice and descriptive")
+          .type("My expression")
+          .blur();
+
+        cy.button("Done").should("be.enabled").click();
+      });
 
-  it("works with a single breakout", () => {
-    const query: StructuredQuery = {
-      "source-table": ORDERS_ID,
-      aggregation: [OFFSET_SUM_TOTAL_AGGREGATION],
-      breakout: [ORDERS_CREATED_AT_BREAKOUT],
-      limit: 5,
-    };
-
-    createQuestion({ query }, { visitQuestion: true });
-
-    verifyNoQuestionError();
-    verifyTableContent([
-      ["April 2022", ""],
-      ["May 2022", "52.76"],
-    ]);
+      cy.log("preview availability");
+      getNotebookStep("data").icon("play").should("be.visible");
+      getNotebookStep("expression").icon("play").should("not.be.visible");
+      getNotebookStep("sort").icon("play").should("be.visible");
+      getNotebookStep("limit").icon("play").should("be.visible");
+
+      visualize();
+      verifyTableContent([
+        ["1", "39.72", ""],
+        ["2", "117.03", "39.72"],
+        ["3", "49.21", "117.03"],
+      ]);
+    });
+
+    it("does not allow to use offset-based column in other clauses (metabase#42764)", () => {
+      const offsettedColumnName = "xyz";
+      const expression = `Offset([${offsettedColumnName}], -1)`;
+      const prefixLength = "Offset([x".length;
+      const prefix = expression.substring(0, prefixLength);
+      const suffix = expression.substring(prefixLength);
+      const query: StructuredQuery = {
+        "source-table": ORDERS_ID,
+        expressions: {
+          [offsettedColumnName]: [
+            "offset",
+            createOffsetOptions(offsettedColumnName),
+            ORDERS_TOTAL_FIELD_REF,
+            -1,
+          ],
+        },
+        "order-by": [["asc", ORDERS_ID_FIELD_REF]],
+        limit: 5,
+      };
+
+      createQuestion({ query }, { visitQuestion: true });
+
+      cy.log("custom column drills");
+      const rowIndex = 1;
+      const columnIndex = 9;
+      const columnsCount = 10;
+      const cellIndex = rowIndex * columnsCount + columnIndex;
+      cy.findAllByRole("gridcell").eq(cellIndex).click();
+      cy.get(POPOVER_ELEMENT).should("not.exist");
```

Comment:
I had to use `POPOVER_ELEMENT` directly because `popover()` has this assertion:

https://github.com/metabase/metabase/blob/784e86989193087bab32e32661393acb766f802a/e2e/support/helpers/e2e-ui-elements-helpers.js#L5-L8